nexus-and-alo/alo-strawman.md

```diff
@@ -0,0 +1,193 @@
+
+You can imagine the minimal Promise inspired ALO API having the following capabilities:
+
+- Start
+- Get Result
+- Get Status (ie: Rejected, Resolved, Fulfilled)
+- Cancellation
+
+The current belief is that ALO should never become a generic Durable Object and therefore it will never be possible to surface arbitrary methods as part of the ALO itself. Instead you would surface those endpoints as either stateless short lived API methods, or as top level ALOs themselves. That being said, there is significant room for extension in the ALO surface area such a Listing & Filtering, and the ability to Pause and Resume.
+
+### Proposal for initial interface of the ALO
+
+```protobuf
+// An ALO is started via a
+message AloInfo {
+  string id = 1;
+  Status status = 2;
+  map<string, string> metadata = 3;
+  repeated string http_result_urls = 4;
+
+  enum Status {
+    STATUS_UNSPECIFIED = 0;
+    RUNNING = 1;
+    COMPLETED = 2;
+  }
+}
+
+service ALOHandler {
+  rpc Start(StartRequest) returns (StartResponse);
+  rpc GetInfo(GetInfoRequest) returns (AloInfo);
+  rpc GetResult(GetResultRequest) (GetResultResponse);
+  rpc Cancel(CancelRequest) returns (CancelRepsonse);
+}
+```
+
+### A bit about Ids
+
+![Lost the client Id](./images/lost-client-id.png)
+
+One of the most common debates around ALOs is whether unique Ids should be a fundamental part of the specification and design. 
+
+Considering ALOs represent the eventual completion of an operation, its impossible to imagine a design where they aren’t identifiable. Note that I did not say “addressable” because technically in the case of asynchronous result delivery it may not be required for the ALO itself to be addressable. That being said, even in the async result delivery case, the result must be delivered with a unique correlation Id. 
+
+![Full client Id flow](./images/full-client-id-flow.png)
+
+Because of this, ALOs will also be required to have a unique Id. Furthermore, because server side generation of unique Ids is inherently lossy, ALO **will be required to be created with a client-side generated unique Id**.
+
+### A non-Temporal example of ALOs
+
+To concretely understand the value of ALOs, it's easiest to showcase how they could be leveraged as part of an HTTP API. The example below will not necessarily be supported initially, and is not using best practices. For instance, the ALOs are all being stored exclusively in memory on a single machine, this means there is no durability and potential race conditions if running multiple servers to handle traffic. 
+
+// TODO (Chad): you may have thoughts on how to improve this
+
+```go
+package main
```

Comment:
IMO this code sample is less useful than fleshing out the above gRPC sample with request/response details would be.